rules/CodingStyle/ClassNameImport/UseImportsTraverser.php

```diff
@@ -77,6 +79,10 @@ private function traverseForType(array $stmts, callable $callable, int $desiredT
                 $this->processGroupUse($node, $desiredType, $callable);
             }
 
+            if (!$node instanceof Namespace_) {
```

Comment:
there are 2 top level node:

- `Rector\Core\PhpParser\Node\CustomNode\FileWithoutNamespace`
- `PhpParser\Node\Stmt\Namespace_`

If on `Namespace_` already don't traverse children, above `GroupUse` and `Use_` no longer traversed.